app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt

```diff
@@ -294,28 +327,67 @@ class StateFragmentPresenter @Inject constructor(
     return fragment.childFragmentManager.findFragmentByTag(TAG_AUDIO_FRAGMENT)
   }
 
-  private fun showHideAudioFragment(isVisible: Boolean) {
-    getStateViewModel().setAudioBarVisibility(isVisible)
+  private fun setAudioFragmentVisible(isVisible: Boolean) {
     if (isVisible) {
-      if (getAudioFragment() == null) {
-        val audioFragment = AudioFragment.newInstance(explorationId, currentStateName)
-        fragment.childFragmentManager.beginTransaction().add(
-          R.id.audio_fragment_placeholder, audioFragment,
-          TAG_AUDIO_FRAGMENT
-        ).commitNow()
-      }
-
-      val currentYOffset = binding.stateRecyclerView.computeVerticalScrollOffset()
-      if (currentYOffset == 0) {
-        binding.stateRecyclerView.smoothScrollToPosition(0)
-      }
+      showAudioFragment()
     } else {
-      if (getAudioFragment() != null) {
-        fragment.childFragmentManager.beginTransaction().remove(getAudioFragment()!!).commitNow()
+      hideAudioFragment()
+    }
+  }
+
+  private fun showAudioFragment() {
+    getStateViewModel().setAudioBarVisibility(true)
+    autoPlayAudio = true
+    (fragment.requireActivity() as AudioButtonListener).showAudioStreamingOn()
+    val currentYOffset = binding.stateRecyclerView.computeVerticalScrollOffset()
+    if (currentYOffset == 0) {
+      binding.stateRecyclerView.smoothScrollToPosition(0)
+    }
+    getAudioFragment()?.let {
+      (it as AudioFragmentInterface).setVoiceoverMappings(explorationId, currentStateName, feedbackId)
+      it.view?.startAnimation(AnimationUtils.loadAnimation(context, R.anim.slide_down_audio))
+    }
+    subscribeToAudioPlayStatus()
+  }
+
+  private fun hideAudioFragment() {
+    (fragment.requireActivity() as AudioButtonListener).showAudioStreamingOff()
+    if (isAudioShowing()) {
+      getAudioFragment()?.let {
+        (it as AudioFragmentInterface).pauseAudio()
+        val animation = AnimationUtils.loadAnimation(context, R.anim.slide_up_audio)
+        animation.setAnimationListener(object : Animation.AnimationListener {
+          override fun onAnimationEnd(p0: Animation?) {
+            getStateViewModel().setAudioBarVisibility(false)
+          }
+          override fun onAnimationStart(p0: Animation?) {}
+          override fun onAnimationRepeat(p0: Animation?) {}
+        })
+        it.view?.startAnimation(animation)
       }
     }
   }
 
+  private fun subscribeToAudioPlayStatus() {
+    val audioFragmentInterface = getAudioFragment() as AudioFragmentInterface
+    audioFragmentInterface.getCurrentPlayStatus().observe(fragment, Observer {
+      if (it == AudioViewModel.UiAudioPlayStatus.COMPLETED) {
+        getAudioFragment()?.let {
+          if (isFeedbackPlaying) {
+            audioFragmentInterface.setVoiceoverMappings(explorationId, currentStateName)
```

Comment:
Should we be passing feedbackId here?

Reply:
Comment is out of date because API changed.